### PR TITLE
docs: add contributor guide page

### PR DESCRIPTION
**Pull request by leo220yuyaodog**
Resolve: #263 
add contributor guide page

### CODE REVIEW

File: docs/contributing.md [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+
+#### Naming convention
+
+e.g.,`signupUrl`for var names,`Signup URL`for UI
+
+#### How to update i18n data? 
+
+Please note that we use [Crowdin](https://crowdin.com/project/casdoor-site) as translating platform and i18next as 
+translating tool. When you add some words using i18next in the `web/` directory, please remember to add what you have 
+added to the `web/src/locales/en/data.json` file. Only **zh** needs to be translated.
```
Comment (Selflocking) [inline docs/contributing.md:143]: why only **zh** needs to be translated.

File: docs/contributing.md [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+Welcome to Casdoor! This document is a guideline about how to contribute to Casdoor.
+
+If you find something incorrect or missing, please leave comments / suggestions.
+
+## Get involved
+
+There are many ways to contribute to Casdoor, here are some ideas to get started:
+
+Use Casdoor and report issues! When using Casdoor, report issues to promote development of Casdoor, no matter bugs or 
+proposal. Before file an issue on GitHub, it would be better to discuss first on Gitter, Casbin Forum, Google group or
```
Comment (Selflocking) [inline docs/contributing.md:18]: plz add gitter link, forum link, qq group link like help page:https://casdoor.org/help

Do we have a google group？
if we have, plz add link.

File: docs/contributing.md [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+translating tool. When you add some words using i18next in the `web/` directory, you can run the `i18n/generate_test.go`
+to auto-generate the `web/src/locales/**/data.json`.
+
+Run `i18n/generate_test.go`:
+```shell
+cd i18n && go test
+```
+
+All languages are filled in English by default. You are encouraged to help translate the newly added 
+strings in the `web/src/locales/zh/data.json` after development.
```
Comment (Selflocking) [inline docs/contributing.md:151]:
```suggestion
strings in the `web/src/locales/zh/data.json` by [Crowdin](https://crowdin.com/project/casdoor-site) after your PR has been merged. 
```

File: docs/contributing.md [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+// limitations under the License.
+```
+
+### Semantic PRs
+
+Your pull requests should follow the Conventional Commits spec. The basic requirement is that only the PR title or at 
+least one commit message. For example, three commonly used PR title are given below:
+
+:::caution
+The PR title must be in lower case.
```
Comment (Selflocking) [inline docs/contributing.md:88]:
```suggestion

The PR title must be in lower case.

```

File: docs/contributing.md [hunk 5]
```diff
@@ -0,0 +1,159 @@
...
+
+3. Create a new branch and work on it
+
+4. Keep your branch in sync
+
+5. Commit your changes (make sure your commit message is concise)
+
+6. Push your commits to your forked repository
+
+7. Create a pull request to **master** branch.
```
Comment (Selflocking) [inline docs/contributing.md:48]:
```suggestion
7. Create a pull request from your branch to our **master** branch.
```

File: docs/contributing.md (hunk 4 above)
Comment (leo220yuyaodog) [reply docs/contributing.md:88]: how to set the font red?

Comment (Selflocking) [reply docs/contributing.md:88]: refer to https://docusaurus.io/docs/markdown-features/admonitions

```mdx
:::danger

Some **content** with _Markdown_ `syntax`. Check [this `api`](#).

:::
```
image

**Pull request merged**